exchanges/poloniex/poloniex_websocket.go

```diff
@@ -28,1128 +30,588 @@ import (
+			})
+		}
+		for i := range resp[x].Bids {
+			if resp[x].Bids[i][1].Float64() <= 0 {
+				continue
+			}
+			bids = append(bids, orderbook.Level{
+				Price:  resp[x].Bids[i][0].Float64(),
+				Amount: resp[x].Bids[i][1].Float64(),
+			})
+		}
 
-	// null returned so ok check is not needed
-	clientOrderID, _ := notification[6].(string)
+		if result.Action == "snapshot" {
+			if err := e.Websocket.Orderbook.LoadSnapshot(&orderbook.Book{
+				Exchange:     e.Name,
+				Pair:         cp,
+				Asset:        asset.Spot,
+				Asks:         asks,
+				Bids:         bids,
+				LastUpdateID: resp[x].LastID,
+				LastUpdated:  resp[x].Timestamp.Time(),
+			}); err != nil {
+				return err
+			}
+			continue
+		}
 
-	e.Websocket.DataHandler <- &order.Detail{
-		Exchange:        e.Name,
-		OrderID:         strconv.FormatFloat(orderID, 'f', -1, 64),
-		Pair:            pair,
-		AssetType:       asset.Spot,
-		Side:            orderSide,
-		Price:           orderPrice,
-		Amount:          orderAmount,
-		RemainingAmount: orderAmount,
-		ClientOrderID:   clientOrderID,
-		Status:          order.Pending,
+		if err := e.Websocket.Orderbook.Update(&orderbook.Update{
+			Pair:       cp,
+			UpdateTime: resp[x].Timestamp.Time(),
+			UpdateID:   resp[x].ID,
+			Asset:      asset.Spot,
+			Action:     orderbook.UpdateOrInsertAction,
+			Asks:       asks,
+			Bids:       bids,
+		}); err != nil {
+			return err
+		}
 	}
 	return nil
 }
 
-func (e *Exchange) processAccountOrderUpdate(notification []any) error {
-	if len(notification) < 5 {
-		return errNotEnoughData
-	}
-
-	orderID, ok := notification[1].(float64)
-	if !ok {
-		return fmt.Errorf("%w order id not float64", errTypeAssertionFailure)
-	}
-
-	a, ok := notification[2].(string)
-	if !ok {
-		return fmt.Errorf("%w amount not string", errTypeAssertionFailure)
-	}
-	amount, err := strconv.ParseFloat(a, 64)
-	if err != nil {
+func (e *Exchange) processTicker(result *SubscriptionResponse) error {
+	var resp []WsTicker
+	if err := json.Unmarshal(result.Data, &resp); err != nil {
 		return err
 	}
-
-	oType, ok := notification[3].(string)
-	if !ok {
-		return fmt.Errorf("%w order type not string", errTypeAssertionFailure)
-	}
-
-	var oStatus order.Status
-	var cancelledAmount float64
-	if oType == "c" {
-		if len(notification) < 6 {
-			return errNotEnoughData
-		}
-		cancel, ok := notification[5].(string)
-		if !ok {
-			return fmt.Errorf("%w cancel amount not string", errTypeAssertionFailure)
-		}
-
-		cancelledAmount, err = strconv.ParseFloat(cancel, 64)
+	tickerData := make([]ticker.Price, len(resp))
+	for x := range resp {
+		cp, err := currency.NewPairFromString(resp[x].Symbol)
 		if err != nil {
 			return err
 		}
-
-		if amount > 0 {
-			oStatus = order.PartiallyCancelled
-		} else {
-			oStatus = order.Cancelled
-		}
-	} else {
-		if amount > 0 {
-			oStatus = order.PartiallyFilled
-		} else {
-			oStatus = order.Filled
+		tickerData[x] = ticker.Price{
+			Last:         resp[x].MarkPrice.Float64(),
```

Comment:
Mark price is not last price

```suggestion
			MarkPrice:         resp[x].MarkPrice.Float64(),
```